src/array.vala

```diff
@@ -2,6 +2,9 @@ using GLib;
 
 public class Vast.Array : Object
 {
+    public const ssize_t NEW_AXIS = ssize_t.MAX;
+    public const size_t DEFAULT_DIMENSION = size_t.MAX;
```

Comment:
The drawback of `axis` is that in cases the axis assignment is non-sense -- hence I added sanity check in `end()` to make sure each original axis is at most used once. (I think we can assert it is used exactly once).